crates/oxc_ecmascript/src/side_effects/may_have_side_effects.rs

```diff
@@ -329,6 +329,10 @@ impl<'a> MayHaveSideEffects<'a> for ArrayExpressionElement<'a> {
                 Expression::ArrayExpression(arr) => arr.may_have_side_effects(ctx),
                 Expression::StringLiteral(_) => false,
                 Expression::TemplateLiteral(t) => t.may_have_side_effects(ctx),
+                Expression::Identifier(ident) => {
+                    // FIXME: we should treat `arguments` outside a function scope to have sideeffects
```

Comment:
This has the same problem with https://github.com/oxc-project/oxc/pull/13114#discussion_r2278683936

Reply:
linking: https://github.com/oxc-project/oxc/issues/4232